app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,102 @@
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: RoleNotFound, InvalidApplicationIdentity
+
+    ValidateApplicationIdentity = CommandClass.new(
+        dependencies: {
+            role_class:                 Role,
+            resource_class:             Resource,
+            application_identity_class: ApplicationIdentity,
+            logger:                     Rails.logger
+        },
+        inputs: %i(account service_id username xms_mirid oid)
+    ) do
+
+      def call
+        parse_xms_mirid
+        # compare xms_mirid with what is defined in annotations
+        validate_application_identity
+      end
+
+      private
+
+      # validate integrity of annotations
+      def application_identity
+        @application_identity ||= @application_identity_class.new(
+            role_annotations: role_annotations,
+            service_id: @service_id
+        )
+      end
+
+      def role_annotations
+        role.annotations
+      end
+
+      # convert host ID to full host ID, cucumber:host:somepolicy
+      # TODO For a Conjur user, an incorrect full id will be returned
+      def role_id
+        @role_id ||= @role_class.roleid_from_username(@account, @username)
+      end
+
+      def role
+        @role ||= @resource_class[role_id]
+        raise SecurityErr::RoleNotFound(role_id) if @username.nil?
+        return @role
```

Comment:
Please change to:
```
raise SecurityErr::RoleNotFound(role_id) if @username.nil?
@role ||= @resource_class[role_id]
```

Reply:
What do you mean? My thought process is as follows:
My goal here is to locate a specific role (`test-policy/vm_sigal` for example). The endpoint is:`/:authenticator/test/cucumber/test-policy/vm_sigal/authenticate`

`role_id` would be `cucumber:test-policy/vm_host` and `@resource_class[cucumber:test-policy/vm_host]` returns the actual resource entry and its data in Conjur

But if `cucumber:test-policy/vm_host` or `@resource_class[cucumber:test-policy/vm_host]` returns nil, then the entity is NotFound

What is wrong with my thought process?

Reply:
just a small nit: for the request `/authenticator/test/cucumber/host/test-policy/vm_sigal/authenticate` the role id will be `cucumber:host:vm_sigal@test-policy`